File: Algorithm/baekjoon/java/src/bj_23057.java
import java.io.InputStreamReader;
import java.util.HashSet;
import java.util.StringTokenizer;
import java.io.BufferedReader;

public class bj_23057 {

    static int N;
    static HashSet<Integer> data = new HashSet<>();
    static int[] arr;

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        N = Integer.parseInt(st.nextToken());
        st = new StringTokenizer(br.readLine());
        int M = 0;

        arr = new int[N];
        for (int i=0; i<N; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
            M += arr[i];
        }

        backtrack(0,0);

        System.out.println((M+1-data.size()));
    }

    private static void backtrack(int idx, int res) {

        data.add(res);

        if (idx==N) return;

        backtrack(idx+1, res+arr[idx]);
        backtrack(idx+1, res);
    }
}

File: Algorithm/baekjoon/bj_23057.py
def backtrack(idx=0, res=0):
    data.add(res)
    if idx == N:
        return
    
    backtrack(idx+1, res+arr[idx])
    backtrack(idx+1, res)
    
    
N = int(input())
arr = [*map(int,input().split())]
M = sum(arr)

data = set() # 여기가 원인인듯

backtrack()
print(M+1-len(data))